packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -576,10 +577,10 @@ class HotRunner extends ResidentRunner {
       if (!(await hotRunnerConfig.setupHotRestart())) {
         return OperationResult(1, 'setupHotRestart failed');
       }
-      result = await _restartFromSources(
+      result = await asyncGuard(() => _restartFromSources(
```

Comment:
Offline we discussed seeing if this one alone is sufficient. Also, we should probably include comments at call sites of `asyncGuard`.

Reply:
Done